src/core/views.py

```diff
@@ -516,16 +536,25 @@ def public_profile(request, uuid):
         is_active=True,
         enable_public_profile=True,
     )
-    roles = models.AccountRole.objects.filter(
-        user=user,
-        journal=request.journal,
-    )
+    if request.journal:
+        roles = models.AccountRole.objects.filter(
+            user=user,
+            journal=request.journal,
+        )
 
-    template = 'core/accounts/public_profile.html'
-    context = {
-        'user': user,
-        'roles': roles,
-    }
+        template = 'core/accounts/public_profile.html'
+        context = {
+            'user': user,
+            'roles': roles,
+        }
+        if not context['roles']:
+            raise Http404()
+    else:
+        template = 'press/staff_profile.html'
```

Comment:
I can't find this template here and it doesn't exist on master. Is it in another PR?

Reply:
Thanks for catching that--I'm going to refactor this view as a part of [#3895](https://github.com/BirkbeckCTP/janeway/pull/3901), so if you would eventually merge that into this, or straight into main, that would be great.

I'm going to make the press and journal views for staff and editorial groups all point to the same profile template, for simplicity and consistency.

Reply:
This has now been done on https://github.com/BirkbeckCTP/janeway/pull/3901